sql/analyzer/pushdown.go

```diff
@@ -425,6 +425,11 @@ func convertFiltersToIndexedAccess(
 			if _, ok := c.Node.(*plan.Limit); ok {
 				return false
 			}
+			if p, ok := c.Node.(*plan.Project); ok {
```

Comment:
Where's the test?

Reply:
I think it's only possible to trigger this condition with new name res. I made more projection nodes and call `pruneCols`, `eraseProjections` in a different order